pyanaconda/modules/common/structures/partitioning.py

```diff
@@ -18,10 +18,232 @@
-from pyanaconda.dbus.structure import DBusData
+from pyanaconda.core.constants import DEFAULT_AUTOPART_TYPE
+from pyanaconda.dbus.structure import DBusData, generate_string_from_data
 from pyanaconda.dbus.typing import *  # pylint: disable=wildcard-import
 
-__all__ = ["MountPointRequest"]
+__all__ = ["PartitioningRequest", "MountPointRequest"]
+
+
+class PartitioningRequest(DBusData):
+    """Partitioning request data."""
+
+    def __init__(self):
+        self._partitioning_scheme = DEFAULT_AUTOPART_TYPE
+        self._file_system_type = ""
+        self._excluded_mount_points = []
+
+        self._encrypted = False
+        self._passphrase = ""
+        self._cipher = ""
+        self._luks_version = ""
+
+        self._pbkdf = ""
+        self._pbkdf_memory = 0
+        self._pbkdf_time = 0
+        self._pbkdf_iterations = 0
+
+        self._escrow_certificate = ""
+        self._backup_passphrase_enabled = False
+
+    @property
+    def partitioning_scheme(self) -> Int:
+        """The partitioning scheme.
+
+        Allowed values:
+            0  Create regular partitions.
+            1  Use the btrfs scheme.
+            2  Use the lvm scheme.
+            3  Use the lvm thinp scheme.
+
+        :return: an number of the partitioning scheme
+        """
+        return self._partitioning_scheme
+
+    @partitioning_scheme.setter
+    def partitioning_scheme(self, scheme: Int):
+        self._partitioning_scheme = scheme
+
+    @property
+    def file_system_type(self) -> Str:
+        """Type of a file system used on the partitions.
+
+        For example: ext4
+
+        :return: a name of a file system type
+        """
+        return self._file_system_type
+
+    @file_system_type.setter
+    def file_system_type(self, value: Str):
+        self._file_system_type = value
+
+    @property
+    def excluded_mount_points(self) -> List[Str]:
+        """Excluded mount points.
+
+        Don't create partitions for the excluded
+        mount points during the partitioning.
+
+        For example: /home, /boot, swap
+
+        :return: a list of mount points
+        """
+        return self._excluded_mount_points
+
+    @excluded_mount_points.setter
+    def excluded_mount_points(self, mount_points: List[Str]):
+        self._excluded_mount_points = mount_points
+
+    @property
+    def encrypted(self) -> Bool:
+        """Should devices be encrypted?
+
+        :return: True or False
+        """
+        return self._encrypted
+
+    @encrypted.setter
+    def encrypted(self, encrypted: Bool):
+        self._encrypted = encrypted
+
+    @property
+    def passphrase(self) -> Str:
+        """Passphrase for all encrypted devices.
+
+        :return: a string with the passphrase
+        """
+        return self._passphrase
+
+    @passphrase.setter
+    def passphrase(self, value: Str):
+        self._passphrase = value
+
+    @property
+    def cipher(self) -> Str:
+        """Encryption algorithm used to encrypt the filesystem.
```

Comment:
Again I wonder if we want to specify somehow what are the valid values - maybe even by pointing to docs ?

Reply:
Well, Blivet will fail at some point if you specify an invalid value. I don't know where to find the valid values, even the kickstart documentation doesn't mention that.

Reply:
Maybe ask @vojtechtrefny about the list of valid values?

Reply:
Unfortunately we don't have a list of valid ciphers here. We just take the string and give it to cryptsetup. List of actually supported combinations of cipher, mode and iv depends on kernel support. Some examples are available here -- https://gitlab.com/cryptsetup/cryptsetup/wikis/DMCrypt